omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js

```diff
@@ -100,11 +101,21 @@ export default class ObsAcrossEncountersController {
           this.config.JSDateFormat,
           this.config.language)
       });
-      for (var uuid of Object.keys(this.conceptsMap)) {
-        row.push({
-          value: (encounter.obs[uuid] ? this.getObsValue(encounter.obs[uuid]) : '') || '',
-          className: this.isRetired(encounter.obs[uuid]) ? 'retiredConcept' : ''
-        });
+      for (let uuid of Object.keys(this.conceptsMap)) {
+        let obsArray = [];
+         if (angular.isArray(encounter.obs[uuid])) {
+          for (let obs of encounter.obs[uuid]) {
+             obsArray.push({
+               value: (obs ? this.getObsValue(obs) : '') || '',
+               className: this.isRetired(obs) ? 'retiredConcept' : ''
+              });
+            }
+        } else {
+           obsArray.push({
+             value: ''
+           });
+         }
```

Comment:
What's this about?

Reply:
We'll need this to maintain previous behavior: When `encounter.obs[uuid]` is null, blank string must be assigned to the cell. I think it's to cope with situations where a configured concept is not present within the encounter. Anyway, not putting the empty string would end up with errors on unit tests as they are now.

This was the previous code:
```
      for (var uuid of Object.keys(this.conceptsMap)) {
        row.push({
          value: (encounter.obs[uuid] ? this.getObsValue(encounter.obs[uuid]) : '') || '',
          className: this.isRetired(encounter.obs[uuid]) ? 'retiredConcept' : ''
        });
      }
```